platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java

```diff
@@ -290,7 +290,7 @@ public DefaultStateManagementComponent(
      * @param signedState the newly complete signed state
      */
     private void stateHasEnoughSignatures(final SignedState signedState) {
-        if (signedState.isStateToSave()) {
+        if (signedState.isStateToSave() && !signedState.hasStateBeenSavedToDisk()) {
```

Comment:
It would be functionally the same

I put it here, because we're making the `signedState.isStateToSave` check here (the `isStateToSave` check could *also* be moved down to `SignedStateFileManager.saveSignedStateToDisk()`).

I don't feel too strongly either way, so I can move it if you prefer

Reply:
Hmm, makes sense. I'm ok if you merge this as is.